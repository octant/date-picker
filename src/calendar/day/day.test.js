/* eslint-env jest */

import React from 'react'
import Day from './'
import renderer from 'react-test-renderer'
import serializer from 'jest-glamor-react'
import Enzyme, {mount} from 'enzyme'
import Adaptor from 'enzyme-adapter-react-16'

Enzyme.configure({adapter: new Adaptor()})
expect.addSnapshotSerializer(serializer)

describe('<Day />', () => {
  test('renders the same each time', () => {
    const component = renderer.create(
      <Day date={new Date(2018, 2, 23)} />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('displays proper date', () => {
    const day = mount(<Day date={new Date(2018, 2, 2)} />)
    expect(day.text()).toEqual('2')
  })

  test('calls the clickMethod when clicked', () => {
    const clickMethod = jest.fn()
    const day = mount(
      <Day
        date={new Date(2018, 2, 2)}
        clickMethod={clickMethod} />
    )
    day.simulate('click')

    expect(clickMethod.mock.calls.length).toBe(1)
  })

  test('sets an id in state', () => {
    const day = mount(
      <Day date={new Date(2018, 2, 2)} />
    )

    expect(day.state().id).toEqual('2018-03-02')
  })
})
