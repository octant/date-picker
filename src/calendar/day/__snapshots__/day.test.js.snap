// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day renders the same 1`] = `
.css-0,
[data-css-0] {
  position: relative;
  height: 2.5em;
  width: 2.5em;
  margin: 0.125em;
  line-height: 2.5em;
  text-align: center;
  background-color: orangered;
}

.css-0:hover,
[data-css-0]:hover {
  background-color: darkblue;
}

<div
  className="css-0"
>
  22
</div>
`;
